scripts/data/alloprof/create_data_retrieval.py

```diff
@@ -118,6 +136,32 @@ def get_text(row):
     alloprof_docs["uuid"].tolist()
 ), "Some relevant document of queries are not present in the corpus"
 
+# convert to Dataset
+alloprof_queries = datasets.Dataset.from_pandas(alloprof_queries)
+alloprof_docs = datasets.Dataset.from_pandas(alloprof_docs)
+
+# identify duplicate documents
+# (duplicates are actually error documents,
+# such as "fiche en construction", " ", ...
+duplicate_docs = Counter(alloprof_docs["text"])
+duplicate_docs = {k:v for k,v in duplicate_docs.items() if v > 1}
+
+# for each text that is in duplicate...
+for dup_text in duplicate_docs:
+    # ...get the ids of docs that have that text
+    duplicate_ids = [d["uuid"] for d in alloprof_docs if d["text"] == dup_text]
+    # ...delete all the documents that have these ids from the corpus dataset
+    alloprof_docs = alloprof_docs.filter(lambda x: x["uuid"] not in duplicate_ids)
+    # ...delete them from the relevant documents in queries
+    alloprof_queries = alloprof_queries.map(lambda x: {"relevant": [i for i in x["relevant"] if i not in duplicate_ids]})
```

Comment:
Why do you remove duplicated texts twice ? (cf. line 79)

Reply:
Dans un cas j'enlève les documents qui sont en duplicat. Dans l'autre cas j'enlève les queries en duplicat